docs/en/Concurrency-Check.md

```diff
@@ -1,3 +1,106 @@
 ## Concurrency Check
 
-TODO
\ No newline at end of file
+### Introduction
+
+Concurrency Check (also known as **Concurrency Control**) refers to specific mechanisms used to ensure data consistency in presence of concurrent changes (multiple processes or users access or change the same data in a database at the same time).
+
+There are two commonly used concurrency control mechanisms/approaches;
+* **Optimistic Locking**: Optimistic Locking is a concurrency control method that does not use record locking. Instead, allows multiple users to attempt to **update** or **delete** the same record without informing the users that others are also attempting to **update/delete** the record. 
```

Comment:
`Optimistic Locking` -> `Optimistic concurrency control`. There is no locking at all.